test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -52,30 +76,436 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+				deleteTestPod(testpod)
+			})
 
-			By("checking the cpuset has balancing disabled")
-			cmd = []string{"cat", "/rootfs/sys/fs/cgroup/cpuset/ovs.slice/cpuset.sched_load_balance"}
-			ovsLoadBalancing, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(ovsLoadBalancing).To(Equal("0"))
+			It("[test_id:64102] Create and remove gu pods to verify affinity of ovs are changed appropriately", func() {
+				cpuID := onlineCPUSet.UnsortedList()[0]
+				smtLevel := nodes.GetSMTLevel(cpuID, workerRTNode)
+				if smtLevel < 2 {
+					Skip(fmt.Sprintf("designated worker node %q has SMT level %d - minimum required 2", workerRTNode.Name, smtLevel))
+				}
+				var testpod1, testpod2 *corev1.Pod
+				var err error
+
+				// Create testpod1
+				testpod1 = pods.GetTestPod()
+				testpod1.Namespace = testutils.NamespaceTesting
+				testpod1.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2"),
+						corev1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				}
+
+				testpod1.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNode.Name}
+				err = testclient.Client.Create(context.TODO(), testpod1)
+				Expect(err).ToNot(HaveOccurred())
+				testpod1, err = pods.WaitForCondition(client.ObjectKeyFromObject(testpod1), corev1.PodConditionType(corev1.PodReady), corev1.ConditionTrue, 5*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod1.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+
+				tasksetcmd := []string{"taskset", "-pc", "1"}
+				testpod1Cpus, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod1, tasksetcmd)
+				testlog.Infof("%v pod is using %v cpus", testpod1.Name, string(testpod1Cpus))
+
+				// Create testpod1
+				testpod2 = pods.GetTestPod()
+				testpod2.Namespace = testutils.NamespaceTesting
+				testpod2.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2"),
+						corev1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				}
+				testpod2.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNode.Name}
+				err = testclient.Client.Create(context.TODO(), testpod2)
+				Expect(err).ToNot(HaveOccurred())
+				testpod2, err = pods.WaitForCondition(client.ObjectKeyFromObject(testpod2), corev1.PodConditionType(corev1.PodReady), corev1.ConditionTrue, 5*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod1.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+
+				By("Getting the container cpuset.cpus cgroup")
+				testpod2Cpus, err := pods.ExecCommandOnPod(testclient.K8sClient, testpod2, tasksetcmd)
+				testlog.Infof("%v pod is using %v cpus", testpod2.Name, string(testpod2Cpus))
+
+				// Get cpus used by the ovnkubenode-pods containers
```

Comment:
The cpu manager race again.

Reply:
@MarSik  i am not understanding the race condition here,  Because In this test case , we are not rebooting the host, we are just creating/deleting gu pods, 
The sequence of events are:
Before the test begins,  node already has performance profile
1. Create a gu pod testpod1
2. Get cpus used by testpod1 using taskset
3. Create a gu pod testpod2
4. Get cpus used by testpod2 using taskset
5. Get cpus used by ovs-vswitchd  (Here we do wait for 30 seconds before we fetch the cpus used by ovs-vswitchd )
6. Delete testpod1
7. Get cpus used by ovs-vswitchd 
8. We verify the cpuid's used by ovs-vswitchd is updated appropriately.   (Here we do wait for 30 seconds before we fetch the cpus used by ovs-vswitchd )